[NXP][Zephyr] Add experimental Ethernet support to NXP zephyr platforms

axelnxp opened the pull request:
* Enable Ethernet for RW61x platform
* Add ethernet support in following examples:
  * All-clusters-app
  * Laudry washer
  * Thermostat
* Support ethernet build with build_examples.py and add it to workflows

axelnxp committed:
[NXP][Zephyr] Add experimental Ethernet support to nxp zephyr platforms

* Enable Ethernet for RW61x platform
* Add ethernet support in following examples:
  * All-clusters-app
  * Laudry washer
  * Thermostat
* Support ethernet build with build_examples.py and add it to workflows

Signed-off-by: Axel Le Bourhis <axel.lebourhis@nxp.com>

restyled-commits committed:
Restyled by gn

marius-alex-tache commented on .github/workflows/examples-nxp.yaml:
```diff
@@ -290,12 +290,13 @@ jobs:
                       --target nxp-rw61x-zephyr-all-clusters \
                       --target nxp-rw61x-zephyr-thermostat \
                       --target nxp-rw61x-zephyr-laundry-washer-factory \
+                      --target nxp-rw61x_eth-zephyr-thermostat-ethernet \
```
Is this a typo `rw61x_eth`?

axelnxp replied:
it's not, I added a new board `rw61x_eth` because to get ethernet support, you need to make hardware modifications. So, in zephyr, we see it as a board variant. I wanted to reflect this here.

marius-alex-tache replied:
Nevermind, I saw the changes below.

axelnxp committed:
[Telink] Define CHIP_DEVICE_CONFIG_ENABLE_ETHERNET as required by shared Zephyr platform

Like nrfconnect platform, define CHIP_DEVICE_CONFIG_ENABLE_ETHERNET to 0
as required by share Zephyr platform files.

Signed-off-by: Axel Le Bourhis <axel.lebourhis@nxp.com>
